include/deal.II/fe/fe_p1nc.h

```diff
@@ -0,0 +1,316 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__fe_p1nc_h
+#define dealii__fe_p1nc_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/fe/fe.h>
+#include <deal.II/base/quadrature.h>
+#include <deal.II/base/qprojector.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/*!@addtogroup fe */
+/*@{*/
+
+/**
+ * Implementation for the scalar version of the P1 nonconforming finite
+ * element, a piecewise linear element on quadrilaterals in 2D.
+ *
+ * Unlike any continuous conforming finite element belongs to H^1_0,
+ * the P1 nonconforming element does not enforce the continuity across edges.
+ * But it requires the continuity just in integral sense:
+ * any function in the space should have the same integral values on two sides of the common edge shared by two adjacent elements.
```

Comment:
Please break this line and the following ones. They are too long to be seen in the diff.